internal/envoy/envoy.go

```diff
@@ -76,6 +82,25 @@ func NewServer(src config.Source, grpcPort, httpPort string) (*Server, error) {
 		envoyPath = "envoy"
 	}
 
+	fullEnvoyPath, err := exec.LookPath(envoyPath)
+	if err != nil {
+		return nil, fmt.Errorf("no envoy binary found: %w", err)
+	}
+
+	// Checksum is written at build time, if it's not empty we verify the binary
+	if Checksum != "" {
```

Comment:
Should we error pretty loudly if there is no checksum set?

Reply:
Extremely nitpicky, but let's do info level if we do that.  It's not an error - the binary was built that way.

In conjunction with https://github.com/pomerium/pomerium/pull/1908/files#r578036674 I'd say we just info log regardless, including if there's no checksum we're validating.